region/src/main/java/org/motechproject/nms/region/service/impl/LocationServiceImpl.java

```diff
@@ -66,89 +75,131 @@ public LocationServiceImpl(StateDataService stateDataService, DistrictService di
         this.healthSubFacilityService = healthSubFacilityService;
     }
 
+
+    private boolean isValidID(final Map<String, Object> map, final String key) {
```

Comment:
A valid id is a long > than 0 or any string other than "0"?  Does this need to be a more involved check?  Can an Integer ever come in?  Should we use a regex that verifies it only contains 0-9.  Or is there no need to do those extra checks.